tests/numpy/test_fill_values.py

```diff
@@ -0,0 +1,61 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import hypothesis.strategies as st
+from hypothesis import given
+from tests.common.debug import minimal, find_any
+from hypothesis.extra.numpy import arrays
+
+
+@given(arrays(object, 100, st.lists(max_size=0)))
+def test_generated_lists_are_distinct(ls):
+    assert len({id(x) for x in ls}) == len(ls)
```

Comment:
I'd prefer an explicit set constructor here over a set comprehension, for clarity.